mule-extensions-api/src/main/java/org/mule/runtime/extension/api/property/MetadataKeyPartModelProperty.java

```diff
@@ -35,18 +38,23 @@ public MetadataKeyPartModelProperty(int order) {
     this(order, true);
   }
 
+  public MetadataKeyPartModelProperty(int order, boolean providedByKeyResolver) {
+    this(order, providedByKeyResolver, false);
+  }
+
   /**
    * Creates a new instance.
    *
    * @param order                 the order of the parameter in the {@link MetadataKey};
    * @param providedByKeyResolver whether or not this part will be provided by the keys resolver associated to the container of
    *                              this part
    */
-  public MetadataKeyPartModelProperty(int order, boolean providedByKeyResolver) {
+  public MetadataKeyPartModelProperty(int order, boolean providedByKeyResolver, boolean isExpressionAllowed) {
```

Comment:
use org.mule.runtime.api.meta.ExpressionSupport instead of the boolean

Reply:
Not sure if we really want to do that. This is not the same concept I think. In fact, we have the information available of whether the expression is supported/required on the parameter. This only refers to whether the expression is valid as a metadata key value. Distinguishing between required/supported doesn't look valid here (that's for the parameter to determine).

Reply:
As discussed in slack, we want to still change it to the ExpressionSupport enum just in case we need it in the future.
I already pushed it. Marking as resolved.